src/shipit_code_coverage/shipit_code_coverage/codecov.py

```diff
@@ -215,26 +236,45 @@ def go(self):
             clone_future = executor.submit(lambda: self.clone_mozilla_central(self.revision))
             clone_future.add_done_callback(lambda f: self.build_files())
 
-        if self.gecko_dev_user is not None and self.gecko_dev_pwd is not None:
-            self.update_github_repo()
-
-        commit_sha = self.get_github_commit(self.revision)
-        logger.info('GitHub revision', revision=commit_sha)
-
-        # TODO: Process suites in parallel.
-        # While we are uploading results for a suite, we can start to process the next one.
-        # TODO: Reenable when Coveralls and/or Codecov will be able to properly handle the load.
-        '''for suite in self.suites:
-            output = self.generate_info(commit_sha, self.coveralls_token, suite)
+        if self.from_pulse:
+            if self.gecko_dev_user is not None and self.gecko_dev_pwd is not None:
+                self.update_github_repo()
 
-            logger.info('Suite report generated', suite=suite)
+            commit_sha = self.get_github_commit(self.revision)
+            logger.info('GitHub revision', revision=commit_sha)
 
-            uploader.coveralls(output)
-            uploader.codecov(output, commit_sha, self.codecov_token, [suite.replace('-', '_')])'''
+            output = self.generate_info(commit_sha, self.coveralls_token)
+            logger.info('Report generated successfully')
 
-        output = self.generate_info(commit_sha, self.coveralls_token)
-        logger.info('Report generated successfully')
-
-        with ThreadPoolExecutorResult(max_workers=2) as executor:
-            executor.submit(lambda: uploader.coveralls(output))
-            executor.submit(lambda: uploader.codecov(output, commit_sha, self.codecov_token))
+            with ThreadPoolExecutorResult(max_workers=2) as executor:
+                executor.submit(lambda: uploader.coveralls(output))
+                executor.submit(lambda: uploader.codecov(output, commit_sha, self.codecov_token))
+        else:
+            try:
+                os.mkdir('code-coverage-reports')
+            except OSError as e:
+                if e.errno != errno.EEXIST:
+                    raise e
+
+            def generate_suite_report(suite):
+                output = self.generate_info(self.revision, 'token', suite, 'lcov')
+
+                self.generate_report(output, suite)
+
+                run_check(['tar', '-cjf', 'code-coverage-reports/%s.tar.bz2' % suite, suite])
+
+                logger.info('Suite report generated', suite=suite)
+
+            for suite in self.suites:
+                with ThreadPoolExecutorResult() as executor:
+                    executor.submit(lambda: generate_suite_report(suite))
+
+            os.chdir('code-coverage-reports')
+            run_check(['git', 'config', '--global', 'http.postBuffer', '12M'])
```

Comment:
Why not generate a `.gitconfig` file here ,

Reply:
What's the benefit compared to using `git config`?